bundles/org.openhab.binding.vektiva/src/main/java/org/openhab/binding/vektiva/internal/handler/VektivaSmarwiHandler.java

```diff
@@ -0,0 +1,263 @@
+            logger.trace("Response: {}", resp.getContentAsString());
+            if (resp.getStatus() == 200) {
+                if (ThingStatus.ONLINE != getThing().getStatus()) {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            } else {
+                updateStatus(ThingStatus.OFFLINE);
+            }
+            return resp.getContentAsString();
+        } catch (InterruptedException e) {
+            logger.debug("API execution has been interrupted", e);
+            updateStatus(ThingStatus.OFFLINE);
+            Thread.currentThread().interrupt();
+        } catch (TimeoutException e) {
+            logger.debug("Timeout during API execution", e);
+            updateStatus(ThingStatus.OFFLINE);
+        } catch (ExecutionException e) {
+            logger.debug("Exception during API execution", e);
+            updateStatus(ThingStatus.OFFLINE);
+        }
+        return null;
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(VektivaSmarwiConfiguration.class);
+        logger.debug("IP address: {}", config.ip);
+
+        future = scheduler.scheduleWithFixedDelay(this::checkStatus, 0, config.refreshInterval, TimeUnit.SECONDS);
+    }
+
+    private synchronized void initializeWebSocketSession() {
+        if (config.useWebSockets) {
+            closeSession();
+            session = createSession();
+            if (session != null) {
+                logger.debug("WebSocket connected!");
+            }
+        }
+    }
+
+    private void checkStatus() {
+        String url = "http://" + config.ip + "/statusn";
+
+        try {
+            ContentResponse resp = httpClient.newRequest(url).method(HttpMethod.GET).send();
+            logger.debug("status values: {}", resp.getContentAsString());
+            if (resp.getStatus() == 200) {
+                processStatusResponse(resp.getContentAsString());
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "got response code: " + resp.getStatus());
+            }
+            // reconnect web socket if not connected
+            if (config.useWebSockets && (session == null || !session.isOpen())
+                    && ThingStatus.ONLINE == getThing().getStatus()) {
+                logger.debug("Initializing WebSocket session");
+                initializeWebSocketSession();
+                return;
+            }
+        } catch (InterruptedException e) {
+            logger.debug("API execution has been interrupted", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "exception during status update");
+            Thread.currentThread().interrupt();
+        } catch (TimeoutException e) {
+            logger.debug("Timeout during status update", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "exception during status update");
```

Comment:
Please provide details such as you do in the log message

Reply:
done